doc/migrating-0.3.0.md

```diff
@@ -0,0 +1,94 @@
+# Migrating To Petastorm 0.3.0
+
+Petastorm 0.3.0 has some breaking changes from previous versions. These include:
+- Change of `Reader` argument order, making `dataset_url` required and first
+- Deprecation of `Sequence` in favor of `NGram`
+- Using `materialize_dataset` when writing out a dataset as opposed to the deprecated/removed `add_dataset_metadata` function
+
+## To change Reader argument order
+This is the simplest breaking change in 0.3.0. Simply change the argument order from:
+```python
+from petastorm.reader import Reader
+reader = Reader(schema_fields, dataset_url)
+```
+To:
+```python
+from petastorm.reader import Reader
+reader = Reader(dataset_url, schema_fields=schema_fields)
+```
+
+## To change from Sequence to NGram
+
+Using Sequence
+```python
+from petastorm.reader import Reader
+from petastorm.sequence import Sequence
+
+sequence = Sequence(length=5, delta_treshold=10, timestamp_field='timestamp')
+schema_fields = MySchema.fields
+reader = Reader(dataset_url=dataset_url, schema_fields=schema_fields, sequence=sequence)
+...
+```
+
+Equivalent Results Using NGram
+```python
+from petastorm.reader import Reader
+from petastorm.ngram import NGram
+
+ngram_fields = NGram(
+  fields={step: MySchema.fields for step in range(5)},
+  delta_treshold=10,
+  timestamp_field='timestamp'
+)
+reader = Reader(dataset_url=dataset_url, sequence=sequence)
```

Comment:
Remove `dataset_url=`?

Also, how is the `ngram_fields` used to initialize Reader?